httpclient/src/main/java/org/apache/http/client/utils/URIUtils.java

```diff
@@ -129,7 +129,8 @@ public static URI createURI(
     public static URI rewriteURI(
             final URI uri,
             final HttpHost target,
-            final boolean dropFragment) throws URISyntaxException {
+            final boolean dropFragment,
```

Comment:
@cstamas Is there any chance you could avoid using multiple boolean parameters and replace them with an enum?

Reply:
Ack, that would really pave the path to have more versatile uri utils.... will do

Reply:
Done in 7c7cdc4800d439e87f0bc8685d91940ecf02b640